### PR TITLE
Slight cleanup to code base 

### PR DESCRIPTION
**Pull request by sabeegrewal**
Slight cleanup to the code base. The following changes are made:
- cleaning up progress tracking files 
- updated scalastyle from 0.8.0 to 1.0.0
- made minor style updates (for example: removed semicolons, unneeded parentheses, incomplete javadoc comments, etc.) 
- updated pom.xml to remove some build warnings

### CODE REVIEW

**Comment by sabeegrewal** [comment]
@sjkwak is added to the review. #Closed

**Comment by sabeegrewal** [comment]
@CodingCat is added to the review. #Closed

File: core/src/main/scala/org/apache/spark/eventhubscommon/client/RestfulEventHubClient.scala [hunk 1]
```diff
@@ -45,7 +45,7 @@ private[spark] class RestfulEventHubClient(
     eventHubNamespace: String,
     numPartitionsEventHubs: Map[String, Int],
     consumerGroups: Map[String, String],
-    policyKeys: Map[String, Tuple2[String, String]],
+    policyKeys: Map[String, String Tuple2 String],
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/client/RestfulEventHubClient.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/client/RestfulEventHubClient.scala [hunk 2]
```diff
@@ -84,7 +84,7 @@ private[spark] class RestfulEventHubClient(
 
   private def aggregateResults[T](undergoingRequests: List[Future[(EventHubNameAndPartition, T)]]):
       Option[Map[EventHubNameAndPartition, T]] = {
-    Await.ready(Future.sequence(undergoingRequests), 60 seconds).value.get match {
+    Await.ready(Future.sequence(undergoingRequests), 60.seconds).value.get match {
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/client/RestfulEventHubClient.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/PathTools.scala [hunk 3]
```diff
@@ -17,46 +17,54 @@
...
 package org.apache.spark.eventhubscommon.progress
 
+import org.apache.hadoop.fs.Path
+
 import org.apache.spark.eventhubscommon.EventHubNameAndPartition
 
 private[spark] object PathTools extends Serializable {
 
-  private def fromSubDirNamesToString(subDirs: Seq[String]): String = {
+  private def combineDirectoryNames(subDirs: Seq[String]): String =
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/PathTools.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 4]
```diff
@@ -30,41 +30,36 @@ import org.apache.hadoop.fs._
 import org.apache.spark.eventhubscommon.{EventHubNameAndPartition, EventHubsConnector, OffsetRecord}
 import org.apache.spark.internal.Logging
 
-private[spark] abstract class ProgressTrackerBase[T <: EventHubsConnector](
+private[spark]
+abstract class ProgressTrackerBase[T <: EventHubsConnector](
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 5]
```diff
@@ -30,41 +30,36 @@ import org.apache.hadoop.fs._
...
-  protected lazy val progressTempDirStr: String = PathTools.progressTempDirPathStr(progressDir,
-    appName)
-  protected lazy val progressMetadataDirStr: String = PathTools.progressMetadataDirPathStr(
-    progressDir, appName)
+  private[spark] lazy val progressDirectoryStr =
+    PathTools.makeProgressDirectoryStr(progressDir, appName)
+  private[spark] lazy val tempDirectoryStr =
+    PathTools.makeTempDirectoryStr(progressDir, appName)
+  private[spark] lazy val metadataDirectoryStr =
+    PathTools.makeMetadataDirectoryStr(progressDir, appName)
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 6]
```diff
@@ -30,41 +30,36 @@ import org.apache.hadoop.fs._
...
 
-  protected lazy val progressDirPath = new Path(progressDirStr)
-  protected lazy val progressTempDirPath = new Path(progressTempDirStr)
-  protected lazy val progressMetadataDirPath = new Path(progressMetadataDirStr)
+  private[spark] lazy val progressDirectoryPath = new Path(progressDirectoryStr)
+  private[spark] lazy val tempDirectoryPath = new Path(tempDirectoryStr)
+  private[spark] lazy val metadataDirectoryPath = new Path(metadataDirectoryStr)
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 7]
```diff
@@ -30,41 +30,36 @@ import org.apache.hadoop.fs._
...
-  // is developed for fast initialization of progress tracker and it should (can) be independent
-  // with Spark Streaming checkpoint (the others have to be coordinated with Spark Streaming
-  // checkpoint cleanup to ensure that we have enough support for recovery). By cleaning metadata
-  // in progress tracker, we can ensure that the ProgressTracker can still be quickly initialized
-  // even the user does not enable Spark Streaming checkpoint or the cleanup of checkpoint is
-  // lagging behind
+  /* Metadata is maintained for fast ProgressTracker initialization and can cleaned independently
+     from Spark checkpoint files. We'll clean metadata here to ensure it's cleaned whether or not
+     Spark checkpointing is enabled.
+   */
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 8]
```diff
@@ -30,41 +30,36 @@ import org.apache.hadoop.fs._
...
   private val threadPoolForMetadataClean = new ScheduledThreadPoolExecutor(1)
   protected val metadataCleanupFuture: ScheduledFuture[_] = scheduleMetadataCleanTask()
 
   // getModificationTime is not reliable for unit test and some extreme case in distributed
   // file system so that we have to derive timestamp from the file names. The timestamp can be the
   // logical one like batch 1, 2, 3 and can also be the real timestamp
-  private[spark] def fromPathToTimestamp(path: Path): Long = {
+  private[spark] def fromPathToTimestamp(path: Path): Long =
     path.getName.split("-").last.toLong
-  }
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 9]
```diff
@@ -281,8 +276,8 @@ private[spark] abstract class ProgressTrackerBase[T <: EventHubsConnector](
     var oos: FSDataOutputStream = null
     try {
       // write progress file
-      oos = fs.create(new Path(s"$progressDirStr/${PathTools.progressFileNamePattern(commitTime)}"),
-        true)
+      oos = fs.create(new Path(
+        s"$progressDirectoryPath/${PathTools.makeProgressFileName(commitTime)}"), true)
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala [hunk 10]
```diff
@@ -309,8 +304,8 @@ private[spark] abstract class ProgressTrackerBase[T <: EventHubsConnector](
   private def createMetadata(fs: FileSystem, commitTime: Long): Boolean = {
     var oos: FSDataOutputStream = null
     try {
-      oos = fs.create(new Path(s"$progressMetadataDirStr/" +
-        s"${PathTools.progressMetadataNamePattern(commitTime)}"), true)
+      oos = fs.create(new Path(
+        s"$metadataDirectoryStr/" + s"${PathTools.makeMetadataFileName(commitTime)}"), true)
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/eventhubscommon/progress/ProgressTrackerBase.scala]: why?

File: core/src/main/scala/org/apache/spark/sql/streaming/eventhubs/checkpoint/StructuredStreamingProgressTracker.scala [hunk 11]
```diff
@@ -60,12 +59,12 @@ class StructuredStreamingProgressTracker(
...
   private def initProgressFileDirectory(): Unit = {
-    val fs = progressDirPath.getFileSystem(hadoopConfiguration)
+    val fs = progressDirectoryPath.getFileSystem(hadoopConfiguration)
     try {
-      val progressDirExist = fs.exists(progressDirPath)
+      val progressDirExist = fs.exists(progressDirectoryPath)
       if (progressDirExist) {
         val (validationPass, latestFile) = validateProgressFile(fs)
-        println(s"${latestFile}")
+        println(s"$latestFile")
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/sql/streaming/eventhubs/checkpoint/StructuredStreamingProgressTracker.scala]: please help to remove this, sorry that I forgot about it

File: core/src/main/scala/org/apache/spark/streaming/eventhubs/EventHubDirectDStream.scala [hunk 12]
```diff
@@ -408,5 +408,5 @@ private[eventhubs] class EventHubDirectDStream private[eventhubs] (
 
 private[eventhubs] object EventHubDirectDStream {
   val cleanupLock = new Object
-  var lastCleanupTime = -1L
+  var lastCleanupTime: Long = -1L
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/streaming/eventhubs/EventHubDirectDStream.scala]: why?

File: core/src/main/scala/org/apache/spark/streaming/eventhubs/EventHubsReceiver.scala [hunk 13]
```diff
@@ -104,9 +104,9 @@ private[eventhubs] class EventHubsReceiver(
   private[eventhubs] class EventHubsMessageHandler() extends Runnable {
 
     // The checkpoint interval defaults to 10 seconds if not provided
-    val checkpointInterval = eventhubsParams.getOrElse("eventhubs.checkpoint.interval", "10").
-      toInt * 1000
-    var nextCheckpointTime = System.currentTimeMillis() + checkpointInterval
+    val checkpointInterval: Long =
+      eventhubsParams.getOrElse("eventhubs.checkpoint.interval", "10").toLong * 1000
+    var nextCheckpointTime: Long = System.currentTimeMillis() + checkpointInterval
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/streaming/eventhubs/EventHubsReceiver.scala]: why need explicit type?

File: core/src/main/scala/org/apache/spark/streaming/eventhubs/checkpoint/DfsBasedOffsetStore.scala [hunk 14]
```diff
@@ -19,9 +19,8 @@ package org.apache.spark.streaming.eventhubs.checkpoint
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
 
-import org.apache.spark.internal.Logging
 import org.apache.spark.SparkContext
-
+import org.apache.spark.internal.Logging
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/streaming/eventhubs/checkpoint/DfsBasedOffsetStore.scala]: why?

File: core/src/main/scala/org/apache/spark/streaming/eventhubs/checkpoint/DirectDStreamProgressTracker.scala [hunk 15]
```diff
@@ -36,7 +36,8 @@ import org.apache.spark.eventhubscommon.progress.ProgressTrackerBase
  * @param appName the name of Spark application
  * @param hadoopConfiguration the hadoop configuration instance
  */
-private[spark] class DirectDStreamProgressTracker private[spark](
+private[spark]
+class DirectDStreamProgressTracker private[spark](
```
Comment (CodingCat) [inline core/src/main/scala/org/apache/spark/streaming/eventhubs/checkpoint/DirectDStreamProgressTracker.scala]: why?

File: core/src/test/scala/org/apache/spark/eventhubscommon/EventhubsImplicitsSuite.scala [hunk 16]
```diff
@@ -25,7 +25,7 @@ import org.apache.spark.streaming.{StreamingContext, TestSuiteBase}
 class EventhubsImplicitsSuite
   extends TestSuiteBase with org.scalatest.Matchers with MockitoSugar {
 
-  val ehParams = Map[String, String] (
+  val ehParams: Map[String, String] = Map(
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/eventhubscommon/EventhubsImplicitsSuite.scala]: why explicit type?

File: core/src/test/scala/org/apache/spark/eventhubscommon/utils/SimulatedEventHubs.scala [hunk 17]
```diff
@@ -80,7 +80,7 @@ class TestEventHubsReceiver(
     offsetType: EventHubsOffsetType)
   extends EventHubsClientWrapper {
 
-  val eventHubName = eventHubParameters("eventhubs.name")
+  val eventHubName: String = eventHubParameters("eventhubs.name")
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/eventhubscommon/utils/SimulatedEventHubs.scala]: why?

File: core/src/test/scala/org/apache/spark/eventhubscommon/utils/SimulatedEventHubs.scala [hunk 18]
```diff
@@ -233,7 +233,7 @@ class FragileEventHubClient private extends EventHubClient {
 // ugly stuff to make things checkpointable in tests
 object FragileEventHubClient {
 
-  var callIndex = -1
+  var callIndex: Int = -1
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/eventhubscommon/utils/SimulatedEventHubs.scala]: why?

File: core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsStreamTest.scala [hunk 19]
```diff
@@ -93,7 +93,7 @@ trait EventHubsStreamTest extends QueryTest with BeforeAndAfter
   }
 
   /** How long to wait for an active stream to catch up when checking a result. */
-  val streamingTimeout = 60.seconds
+  val streamingTimeout: Span = 60.seconds
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsStreamTest.scala]: why?

File: core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsStreamTest.scala [hunk 20]
```diff
@@ -131,7 +131,8 @@ trait EventHubsStreamTest extends QueryTest with BeforeAndAfter
         isSorted = false)
     }
 
-    def apply(rows: Row*): CheckAnswerRows = CheckAnswerRows(rows, false, false)
+    def apply(rows: Row*): CheckAnswerRows =
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsStreamTest.scala]: why?

File: core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsStreamTest.scala [hunk 21]
```diff
@@ -279,7 +280,7 @@ trait EventHubsStreamTest extends QueryTest with BeforeAndAfter
       }
     }
 
-    def isStreamWaitingAt(time: Long): Boolean = synchronized {waitStartTime == Some(time)}
+    def isStreamWaitingAt(time: Long): Boolean = synchronized {waitStartTime.contains(time)}
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsStreamTest.scala]: why?

File: core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/checkpoint/StructuredStreamingProgressTrackerSuite.scala [hunk 22]
```diff
@@ -37,26 +37,28 @@ class StructuredStreamingProgressTrackerSuite extends SharedSQLContext {
   }
 
   test("progress directory is created properly when it exists") {
-    fileSystem.mkdirs(new Path(PathTools.progressTempDirPathStr(progressRootPath.toString,
-      appName)))
+    fileSystem.mkdirs(PathTools.makeTempDirectoryPath(progressRootPath.toString, appName))
+
     progressTracker = StructuredStreamingProgressTracker
       .initInstance(eventhubsSource1.uid, progressRootPath.toString, appName, new Configuration())
+
```
Comment (CodingCat) [inline core/src/test/scala/org/apache/spark/sql/streaming/eventhubs/checkpoint/StructuredStreamingProgressTrackerSuite.scala]: why?

**Pull request merged**